### DISCUSSION TITLE
Database replication for multi-read

### DISCUSSION THREAD

**Discussion by michaelwills** [supabase/supabase#10119] [Questions] [answered]
Hello!

I have been looking at ways to allow single write/multi-region reads with supabase. The closest I can find is the discussion here for backups:

[Wal-g backups on free tier, replication connection · Discussion #9272 · supabase/supabase](https://github.com/supabase/supabase/discussions/9272)

The dashboard UI shows publication creation, but that appears to be only for allowing realtime client subscriptions but not for sending data to another database.

Are there any options?

**Comment by burggraf** [comment, marked as answer]
Here's my guide for setting up a read replica (which will require a separate project):  [https://github.com/burggraf/postgresql_replicas](https://github.com/burggraf/postgresql_replicas)

You can set up replicas in multiple regions to speed up worldwide reads (as many as you like).

**Reply by michaelwills** [reply]
Much obliged! The notes on the dashboard SQL editor, realtime slots, and debugging are very helpful, indeed. 👍

**Reply by michaelwills** [reply]
Thanks again @burggraf. I can spin up read replicas now. I added an issue to clarify some issues I hit while implementing it. I'll try to come back around to submit a PR in the next couple/few days.

https://github.com/burggraf/postgresql_replicas/issues/1

**Reply by BoscoDomingo** [reply]
@burggraf Since this is now deprecated as we can't make superusers anymore, and it seemed to fail when schema changes happened, what is the alternative?

I've been looking at 3rd party tools, but they all fail trying to connect or performing actions.

**Reply by CuriousLearner** [reply]
Hello, 

Is there any update on this please?

**Comment by michaelwills** [comment]
@burggraf It has worked well until I had to update the schema. I went through all the steps and one database (Singapore region) is ok, but another in Sydney fails with duplicate key errors in the logs. It's almost like it's attempting to replay again even though there is only one slot active for it. I'll dig again because when it worked it was very nice. :)

**Reply by apristen** [reply]
@michaelwills , 
How you set up Subscription for "slave" Supabase? 
I created Publication(s) at "master" Supabase with no problem using pgAdmin GUI (or even SQL statements), then tried to create Subscription using pgAdmin at "slave" and it says that I need to be superuser to do that. 
And as far as I know, the only superuser is "supabase_admin" which is impossible to use for login :-( 
My user is "postgres".
Any ideas how to create Subscription at Supabase "slave" host?
Thanks in advance!
